test/remake_tests.jl

```diff
@@ -265,3 +265,11 @@ u0 = [1.0; 2.0; 3.0]
 tspan = (0.0, 100.0)
 prob = ODEProblem(lorenz!, u0, tspan, nothing)
 @test_nowarn remake(prob, p = (), interpret_symbolicmap = true)
+
+# IntervalNonlinearProblem doesn't have a u0
+# Issue#726
+interval_f(u, p) = u * u - 2.0 + p[1]
+uspan = (1.0, 2.0)
+prob_int = IntervalNonlinearProblem(interval_f, uspan)
+prob2 = @inferred IntervalNonlinearProblem remake(prob_int; p = [0])
+@test prob2.p == [2]
```

Comment:
```julia
@test prob2.p == [0]
@test prob2.u ≈ √2
```

Reply:
This MWE code is a modified version of the tutorial in the documentation [here](https://docs.sciml.ai/NonlinearSolve/stable/tutorials/getting_started/#Problem-Type-2:-Solving-Interval-Rootfinding-Problems-with-Bracketing-Methods). It does not use `@inferred`.